optaplanner-persistence/optaplanner-persistence-jackson/src/test/java/org/optaplanner/persistence/jackson/api/score/constraint/ConstraintMatchJacksonJsonSerializerAndDeserializerTest.java

```diff
@@ -0,0 +1,82 @@
+package org.optaplanner.persistence.jackson.api.score.constraint;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.Before;
+import org.junit.Test;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConstraintMatchJacksonJsonSerializerAndDeserializerTest {
+
+    ObjectMapper mapper;
+
+    @Before
+    public void setUp() throws Exception {
+        mapper = new ObjectMapper();
+        mapper.registerModule(new SimpleModule()
+                .addSerializer(HardSoftScore.class, new ScoreJacksonJsonSerializer<HardSoftScore>())
+                .addDeserializer(HardSoftScore.class, new HardSoftScoreJacksonJsonDeserializer()));
+    }
+
+    private void testSerializeAndDeserialize(String packageName, String name, Score score, Object... list) {
```

Comment:
the prefix "test" in the name of this method is misleading, as in junit 4 era that were suppose to be test methods (and this doesn't have a @Test annotation because it's not a test)

Reply:
Will rename to `assertSerializeAndDeserialize` to match `AbstractScoreJacksonJsonSerializerAndDeserializerTest` method this is based on.